networkx/algorithms/flow/tests/test_networksimplex.py

```diff
@@ -385,3 +385,180 @@ def test_graphs_type_exceptions():
+        G.add_edge("s5", "s0", weight=1)
+        G.add_edge("s5", "s10", weight=1)
+        G.add_edge("s5", "s11", weight=1)
+        G.add_edge("s5", "s8", weight=1)
+        G.add_edge("s5", "s9", weight=1)
+        G.add_edge("s6", "s0", weight=1)
+        G.add_edge("s6", "s10", weight=1)
+        G.add_edge("s6", "s4", weight=1)
+        G.add_edge("s6", "s6", weight=1)
+        G.add_edge("s7", "s0", weight=1)
+        G.add_edge("s7", "s10", weight=1)
+        G.add_edge("s7", "s4", weight=1)
+        G.add_edge("s7", "s6", weight=1)
+        G.add_edge("s8", "s0", weight=1)
+        G.add_edge("s8", "s11", weight=1)
+        G.add_edge("s8", "s8", weight=1)
+        G.add_edge("s8", "s9", weight=1)
+        G.add_edge("s9", "s0", weight=1)
+        G.add_edge("s9", "s11", weight=1)
+        G.add_edge("s9", "s5", weight=1)
+        G.add_edge("s9", "s7", weight=1)
+        G.add_edge("s9", "s8", weight=1)
+
+        return G
+
+    try:
+        G1 = create_graph(False)
+        flow_value, flow_dict = nx.network_simplex(G1)
+    except Exception as e:
+        pytest.fail(f"network_simplex(G1) failed with error: {e}")
+
+    try:
+        G2 = create_graph(True)
+        flow_value, flow_dict = nx.network_simplex(G2)
+    except Exception as e:
+        pytest.fail(f"network_simplex(G2) failed with error: {e}")
+
+
+def test_case2_issue7562():
+    cap = 1000000000
+
+    def create_graph(with_capacity):
+        G = nx.DiGraph()
+
+        c = {"capacity": cap} if with_capacity else {}
+
+        # Add nodes with demands
+        G.add_node("s0", demand=-89)
+        G.add_node("s1", demand=-197)
+        G.add_node("s2", demand=89)
+        G.add_node("s3", demand=212)
+        G.add_node("s4", demand=32)
+        G.add_node("s5", demand=83)
+        G.add_node("s6", demand=-159)
+        G.add_node("s7", demand=-68)
+        G.add_node("s8", demand=116)
+        G.add_node("s9", demand=120)
+        G.add_node("s10", demand=-65)
+        G.add_node("s11", demand=-74)
+
+        # Add edges
+        G.add_edge("s0", "s1", weight=1, **c)
+        G.add_edge("s1", "s2", weight=1)
+        G.add_edge("s1", "s3", weight=1)
+        G.add_edge("s1", "s4", weight=1)
+        G.add_edge("s10", "s0", weight=1)
+        G.add_edge("s11", "s9", weight=1)
+        G.add_edge("s2", "s5", weight=1)
+        G.add_edge("s5", "s11", weight=1)
+        G.add_edge("s5", "s8", weight=1)
+        G.add_edge("s6", "s10", weight=1)
+        G.add_edge("s7", "s0", weight=1)
+
+        return G
+
+    # Calculate min-cost flow
+    try:
+        G1 = create_graph(False)
+        flow_value, flow_dict = nx.network_simplex(G1)  # fails
+    except Exception as e:
+        pytest.fail(f"network_simplex(G1) failed with error: {e}")
+
+    try:
+        G2 = create_graph(True)
+        flow_value, flow_dict = nx.network_simplex(G2)
+    except Exception as e:
+        pytest.fail(f"network_simplex(G2) failed with error: {e}")
+
+
+def test_case3_issue7562():

```

Comment:
Re: tests, my personal vote would be to keep test_case3 and ditch the other two, which IIUC probe the same behavior.